fiducial_follow/README.md

```diff
@@ -5,25 +5,115 @@
 
 
 ### Paramaters
 
-`target_fiducial`: the fiducial we are following. Default `fid_49`.
+* `target_fiducial`: the fiducial we are following. Use `none` to start idle. Default `fid_49`.
+* `min_dist`: the minimum distance in meters to consider it followed. Default `0.4`.
+* `max_dist`: the maximum distance in meters where we will try to follow. Default `4.5`.
+* `max_linear_rate`: the maximum approach speed used when following in meters per sec. Default `1.0`.
+* `max_angular_rate`: the maximum rotation speed used when following in radians per sec. Default `1.2`.
+* `search_for_target`: rotate to find the fiducial if it is not in view. Default `True`.
+* `lost_angular_rate`: angular rotation rate in radians per sec used in searching. Default `0.6`.
+* `drive_rate`: rate in meters per second for the simple drive commands.  Default `0.4`.
+* `rotate_rate`: rate in radians per second for the simple rotate commands.  Default `1.5`.
 
 ### Publications
 
-`cmd_vel`(geometry_msgs/Twist): commands to move the robot.
+* `cmd_vel` (geometry_msgs/Twist): commands to move the robot.
+* `follower_status` (custom_msgs/FollowerStatus): shows status as commands are received, started, and completed
 
 ### Subscriptions
 
-`fiducial_transforms`:(fiducial_msgs/FiducialTransformArray)
+* `fiducial_transforms` (fiducial_msgs/FiducialTransformArray): Listen for locations of observed fiducials
+* `follower_commands` (custom_msgs/FollowerCommand): Accept commands to process in order of reception
+
+### Example usage for simple single fiducial following
+For the simple legacy fiducial follow mode a single aruco fiducial of size 140mm on edge of the black pattern that is number 49 is used.
+Prepare such a fiducial and place it where the robot can see it with the camera then start the program.
+
+    roslaunch magni_demos fiducial_follow.launch
+
+### Usage for following a sequence of fiducials and/or performing simple movements
+
+The follow.py script can act on commands sent to it using ros topic /follower_commands.
+These commands arrive in a general message, FollowerCommand, upon a well known topic of /follower_commands that is monitored by the follower script. 
+
+A powerful application of the following of a sequence of commands is to have the robot follow fiducials on the floor with a downward facing camera configured for your robot.  For optimal floor fiducial following there are some configuration changes suggested where the config files will be present starting from a path of /opt/ros/kinetic/share/raspicam_node for the case of ROS Kinetic.  Floor fiducials of 140mm size up to almost a 2 meters away have been recognized with success. 
```

Comment:
Confusing sentence.  I read it several times and am still not sure what it means.

Reply:
Ok, I agree, quite a run-on sentence